pydra/engine/submitter.py

```diff
@@ -110,41 +98,37 @@ async def submit(self, runnable, wait=False, rerun=False):
             Coroutines for :class:`~pydra.engine.core.TaskBase` execution.
 
         """
+        if runnable.plugin and runnable.plugin != self.plugin:
+            raise NotImplementedError()
```

Comment:
This was removed since it wasn't being tested.

original:
```python
# dj: this is not tested!!! TODO
await self.worker.run_el(workflow, rerun=rerun)
```